doc/adr/0015-current-API-authentication-strategy.md

```diff
@@ -0,0 +1,116 @@
+# 0015. A description of the current API authentication strategy
+
+## Status
+
+Accepted
+
+## Problem
+
+We have no documentation around how authentication works in Weave Gitops. This will aim to capture the current state of
+the system.
+
+The 'decision' section has been skipped as the decisions have been made and reverse-engineering the rationale for them is unlikely to work. The 'context' section covers implementation details and the 'consequences' aims to enumerate major risks with the system as well as how to work with it safely.
+
+This supersedes the sections of [ADR-0005 Wego Core Auth Strategy](./0005-wego-core-auth-strategy.md) not to do with authentication to GitHub/GitLab.
+
+## Context
+
+**Note:** throughout this I refer to "auth methods" and "methods" which, for the sake of this document, mean the general process of authenticating who a request is coming from NOT the specific object method which may be called.
+
+Since ADR-0005 was accepted there have been several changes to Weave GitOps. The largest of these has been the removal of support for making changes to git repositories and a greater use of `impersonation` for authentication to the Kubernetes API.
+
+```mermaid
+flowchart TD
+  start([Request]) --> iterate["Next authentication method (oidc, user etc.)"]
+
+  auth -.-> extAuth["[Optional] external authentication (e.g. OIDC)"]
+  extAuth -.-> auth
+
+  subgraph WithAPIAuthentication
```

Comment:
what is the system representing this `subgraph` is kuberentes API, is weave gitops, both ?

Reply:
The subgraphs refer to functions and methods in specific packages of weave gitops.

Arguably a lot of this isn't really 'architecture' and is probably a bit low-level for that but I wanted to make sure it was captured because what architecture exists isn't particularly intuitive.

Reply:
(but I'll add some annotations to the graph to make it clear where this is from)